File: 1614.cpp
class Solution {
public:
    int maxDepth(string s) {
        vector<char> stack;
        int result=0;
        int count=0;
        int maxval=0;
        
        for(int i=0;i<s.length();i++)
        {
            if(s[i]=='(')
            {
                stack.push_back('(');
                count++;
                if(maxval<count)
                {
                    maxval=count;
                }
            }
            else if(s[i]==')')
            {
                stack.pop_back();
                count--;
                if(stack.empty())
                {
                    if(result<maxval)
                    {
                        result=maxval;
                    }
                    count=0;
                    maxval=0;
                }
            }
        }
        return result;
    }
};

File: 1614.py
class Solution:
    def maxDepth(self, s: str) -> int:
        stack=list()
        stackcount=list()
        count=0
        maxval=0
        
        for i in s:
            if i=='(':
                stack.append('(')
                count+=1
                if maxval<count:
                    maxval=count
            elif i==')':
                stack.pop()
                count-=1
                if len(stack)==0:
                    stackcount.append(maxval)
                    count=0
                    maxval=0
            else:
                continue
                
        if len(stackcount)==0:
            return 0
        
        return max(stackcount)
                    
            
